pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1116,6 +1147,159 @@ func lookupDomainConfig(ctx *domainContext, key string) *types.DomainConfig {
 	return &config
 }
 
+func getHostCpusNum() (int, error) {
+	// The `nproc` util relies on the /proc/sys file system, that is not properly accessible from a container.
+	// So, have to run the command with access to the host FS.
+	out, err := exec.Command("chroot", "/hostfs", "nproc", "--all").Output()
+	if err != nil {
+		return -1, err
+	}
+	// remove "\n"
+	out = out[:2]
+	maxHostCpus, _ := strconv.Atoi(string(out))
+	return maxHostCpus, nil
+}
+
+func setCgroupCpuset(vmUUIDAndVersion string, cpus string) error {
+	cpuSetFileName := fmt.Sprintf("/sys/fs/cgroup/cpuset/eve-user-apps/%s/cpuset.cpus", vmUUIDAndVersion)
```

Comment:
I am not sure that we will have this path available in case of stopped domain.

Reply:
The function is not called for the deleted Domains. It's used to fix the CPU sets of the still-running domains.

Reply:
We use the containerd/cgroups package now for that.